src/sinks/loki/config.rs

```diff
@@ -16,6 +16,43 @@ use crate::{
     tls::{TlsConfig, TlsSettings},
 };
 
+/// Loki-specific compression.
+#[configurable_component]
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum ExtendedCompression {
+    /// Snappy compression.
+    ///
+    /// This implies sending push requests as Protocol Buffers.
+    #[serde(rename = "snappy")]
+    Snappy,
+}
+
+/// Compose with basic compression and Loki-specific compression.
+#[configurable_component(sink)]
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+#[serde(untagged)]
+pub enum CompressionConfigAdapter {
+    /// Basic compression.
+    Original(#[configurable(derived)] Compression),
+    /// Loki-specific compression.
+    Extended(#[configurable(derived)] ExtendedCompression),
+}
+
+impl CompressionConfigAdapter {
+    pub const fn content_encoding(self) -> Option<&'static str> {
+        match self {
+            CompressionConfigAdapter::Original(compression) => compression.content_encoding(),
+            CompressionConfigAdapter::Extended(_) => None,
```

Comment:
```suggestion
            CompressionConfigAdapter::Extended(_) => Some("snappy"),
```

Shouldn't we be setting the content encoding header to `snappy` for these requests?

Reply:
Just like [Promtail](https://github.com/grafana/loki/blob/main/clients/pkg/promtail/client/client.go#L399-L401), we needn't to set this header. The [comments](https://github.com/grafana/loki/blob/557fdf2206f9942022ce3608952d419185616ab9/pkg/loghttp/push/push.go#L64-L66) from Loki also mentioned:

> Note: HTTP clients do not need to set this header

Reply:
👍 I might still have us set it since it is more correct from an HTTP-spec perspective and Loki appears to handle it correctly if it is set.

Reply:
> 👍 I might still have us set it since it is more correct from an HTTP-spec perspective and Loki appears to handle it correctly if it is set.

Aha, yes we can. I merged your suggestion commit, thanks! @jszwedko